File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.cs
using System;

public class Solution {
    public int solution(int n, int m, int[] section) {
       
        int answer = 1;
        //int mm = section[0] ;
        int mRange =section[0] + m-1;
         for(int i = 0; i < section.Length; i ++ )
         {
             
            if(section[i] > mRange)
            {
                answer++;
                mRange = section[i]+m-1;
                    
            }
         }
        return answer;
    }
}

File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.cpp
#include <string>
#include <vector>
using namespace std;

int solution(int n, int m, vector<int> section) {
    int answer = 0;
    int last = 0; 

    for(int i = 0; i < section.size(); i++) {
        
        if(section[i] > last) {
            answer++; 
            last = section[i] + m - 1; 
        }
    }

    return answer;
}